### PR TITLE
Update resources to support Chef 12.5 and 12.6

### PR DESCRIPTION
**Pull request by iennae**
### Description

Fix cookbook resources to support Chef 12.5 and 12.6 

### Issues Resolved

Issue #147 

### Check List

- [ ] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [ ] New functionality includes testing.
- [ ] New functionality has been documented in the README if applicable
- [ ] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by wrightp** [comment]
The class_eval appears to only be needed for 12.5 and 12.6. I'm guessing this is the preferred workaround which I don't particularly like. We can get this merged to fix the issue for now, but I will probably fix this so we don't class_eval 12.7+.

Please fix linting error and merge. Thank you for helping getting this along.

**Comment by wrightp** [comment]
Fauxhai!!! More to fix apparently.

File: metadata.rb [hunk 1]
```diff
@@ -8,3 +8,8 @@
 source_url 'https://github.com/chef-cookbooks/chef-ingredient'
 issues_url 'https://github.com/chef-cookbooks/chef-ingredient/issues'
 chef_version '>= 12.5' if respond_to?(:chef_version)
+
+supports 'debian'
+supports 'rhel'
+supports 'suse'
+supports 'windows'
```
Comment (schisamo) [inline metadata.rb]: Are these _platforms_ or _platform families_? If it's the former we need to add:

* `centos`
* `freebsd`
* `mac_os_x`
* `solaris2`
* `ubuntu`

If it's the latter then we need to figure out how to represent these platforms as platform families. This cookbook essentially should support everything Chef supports.

Comment (wrightp) [inline metadata.rb]: I was just trying to satisfy the error for the moment. Not the final solution.

**Comment by wrightp** [comment]
I like the metadata "supports" rule, but let's deal with this later. There is still the task of migrating the omnitruck platform mapping to mixlib-install which will remove the OmnitruckHandler. That sounds like a good time to add the supported list of platforms.

**Pull request merged**